sklearn/linear_model/_base.py

```diff
@@ -49,6 +50,77 @@
+# in cases where now normalize=False. The default value of 'normalize' should
+# be changed to False in linear models where now normalize=True
+def _deprecate_normalize(normalize, default, estimator_name):
+    """ Normalize is to be deprecated from linear models and a use of
+    a pipeline with a StandardScaler is to be recommended instead.
+    Here the appropriate message is selected to be displayed to the user
+    depending on the default normalize value (as it varies between the linear
+    models and normalize value selected by the user).
+
+    Parameters
+    ----------
+    normalize : bool,
+        normalize value passed by the user
+
+    default : bool,
+        default normalize value used by the estimator
+
+    estimator_name : string,
+        name of the linear estimator which calls this function.
+        The name will be used for writing the deprecation warnings
+
+    Returns
+    -------
+    normalize : bool,
+        normalize value which should further be used by the estimator at this
+        stage of the depreciation process
+
+    TODO: it should be updated in v1.2.0 and removed completely in v1.4.0
+    """
+    if normalize == 'deprecated':
+        _normalize = default
+    else:
+        _normalize = normalize
+
+    if default and normalize == 'deprecated':
+        warnings.warn(
+            " default of 'normalize' will be set to False in version 1.2.0 and"
+            " deprecated in version 1.4.0"
+            " \nPass normalize=False and use Pipeline with a StandardScaler in"
+            " a preprocessing stage:"
+            "  model = make_pipeline(StandardScaler(),"
+            f" {estimator_name}(normalize=False))"
+            "\nIf you wish to use additional parameters in"
+            " the fit() you can include them as follows:"
+            " kwargs = {model.steps[-1][0] + "
+            "'__<your_param_name>': <your_param_value>}\n"
+            "model.fit(X, y, **kwargs)", FutureWarning
+        )
+    elif normalize != 'deprecated' and normalize and not default:
+        warnings.warn(
+            "'normalize' was deprecated in version 1.0.0 and will be "
+            "removed in 1.2.0 \nIf you still wish to normalize use "
+            "Pipeline with a StandardScaler in a preprocessing stage: "
+            "model = make_pipeline(StandardScaler(), "
+            f"{estimator_name}()). \nIf you wish to use additional "
+            "parameters in the fit() you can include them as follows: "
+            "kwargs = {model.steps[-1][0] + "
+            "'__<your_param_name>': <your_param_value>}\n"
+            "model.fit(X, y, **kwargs)", FutureWarning
+        )
+    elif normalize != 'deprecated' and not normalize and not default:
+        warnings.warn(
+            "'normalize' was deprecated in version 1.0.0 and will be"
+            " removed in 1.2.0 Don't set 'normalize' parameter"
+            " and leave it to its default value", FutureWarning
+        )
```

Comment:
I think `normalize != 'deprecated' and not normalize` is equivalent to `normalize is False`, right?

We should not raise any `FutureWarning` for this case. We don't want to raise warnings for users who do not have to change anything in their code.

Reply:
you are right about the `if` argument, I am updating it to: `if not normalize and not default`
However, in this case `normalize` will no longer exist from version 1.2 so if the user is setting it we need to give them the deprecation message, not to set `normalize` parameter as setting it will cause problems from v1.2

Reply:
Indeed you are correct, I get it know. It's so hard to get this kind of counter-factual reasoning correctly.